src/bin/firebase-functions.ts

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env node
+
+import * as express from 'express';
+import { loadBackend } from '../runtime/loader';
+
+function printUsage() {
+  console.error(
+    `
+Usage: firebase-functions <functionsDir>
+
+Arguments:
+  - functionsDir: Directory containing source code for Firebase Functions.
+`
+  );
+  process.exit(1);
+}
+
+const args = process.argv.slice(2);
+if (args.length != 1) {
+  printUsage();
+}
+const functionsDir = args[0];
+
+let server;
+const app = express();
+
+app.get('/backend.yaml', async (req, res) => {
+  try {
+    const backend = await loadBackend(functionsDir);
+    res.setHeader('content-type', 'text/yaml');
+    res.send(JSON.stringify(backend));
+  } catch (e) {
+    res
+      .status(400)
+      .send(`Failed to generate manifest from function source: ${e}`);
+  }
+});
+
+app.get('/quitquitquit', async (req, res) => {
+  res.send('ok');
+  server.close(() => console.log('shutdown requested via /quitquitquit'));
+});
+
+let port = 8080;
```

Comment:
You're conforming to what's in the CLI now. We never worked around the fact that Python _really_ doesn't want to allow two ports. So instead of having a control API on one port and the main API on another, we'd expose the control API when STACK_CONTROL_API is set. That's what's in the WTF-is-? deck at least. I don't love the name "STACK_CONTROL_API" so feel free to suggest a different one.